src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/BoundedCache.cs

```diff
@@ -0,0 +1,405 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace CoreWCF.IdentityModel
+{
+    /// <summary>
+    /// A cache of type T where items are cached and removed 
+    /// according to the type specified, currently only 'TimeBounded' is supported.  An items is added with an expiration time.
+    /// </summary>
+    internal class BoundedCache<T> where T : class
+    {
```

Comment:
Can you see if this is replaceable with a MemoryCache? An article going over how to use it is [here](https://docs.microsoft.com/en-us/aspnet/core/performance/caching/memory?view=aspnetcore-6.0). Owning less code is better so if we can get rid of this, we should.